src/kdbindings/signal.h

```diff
@@ -242,14 +259,13 @@ class Signal
      * The `KDBindings::Signal` class itself is not thread-safe. While the `ConnectionEvaluator` is inherently
      * thread-safe, ensure that any concurrent access to this Signal is protected externally to maintain thread safety.
      */
-    ConnectionHandle connectDeferred(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(Args...)> const &slot)
+    ConnectionHandle connectDeferred(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(const ConnectionHandle &, Args...)> const &slot)
```

Comment:
The fact that the lambda used by `connectDeferred` is passed a ConnectionHandle is an implementation detail and should not be visible to the user.